engine/crates/engine-v2/schema/src/entity.rs

```diff
@@ -0,0 +1,43 @@
+use walker::{Iter, Walk};
+
+use crate::{DefinitionId, EntityDefinition, EntityDefinitionId, TypeSystemDirective};
+
+impl From<EntityDefinitionId> for DefinitionId {
+    fn from(value: EntityDefinitionId) -> Self {
+        match value {
+            EntityDefinitionId::Interface(id) => DefinitionId::Interface(id),
+            EntityDefinitionId::Object(id) => DefinitionId::Object(id),
+        }
+    }
+}
+
+impl EntityDefinitionId {
+    pub fn maybe_from(definition: DefinitionId) -> Option<EntityDefinitionId> {
+        match definition {
+            DefinitionId::Object(id) => Some(EntityDefinitionId::Object(id)),
+            DefinitionId::Interface(id) => Some(EntityDefinitionId::Interface(id)),
+            _ => None,
+        }
+    }
+
+    pub fn is_object(&self) -> bool {
+        matches!(self, EntityDefinitionId::Object(_))
+    }
+}
+
+impl<'a> EntityDefinition<'a> {
+    pub fn name(&self) -> &'a str {
+        match self {
+            EntityDefinition::Object(item) => item.name(),
+            EntityDefinition::Interface(item) => item.name(),
+        }
+    }
+
+    pub fn directives(&self) -> impl Iter<Item = TypeSystemDirective<'a>> + 'a {
+        let (schema, directive_ids) = match self {
+            EntityDefinition::Object(item) => (item.schema, &item.as_ref().directive_ids),
+            EntityDefinition::Interface(item) => (item.schema, &item.as_ref().directive_ids),
+        };
+        directive_ids.walk(schema)
+    }
```

Comment:
This module looks a lot like the generated one, is it still needed?

Reply:
it's adding a few functions to the generated walker, like `name()`. I found it too risky to complexify the graphql schema even more to auto-generate those, so just writing them by hand

Reply:
and yes all of those are used. Those extra functions are more or less good, but would require reworking the `engine-v2` operation part to change them

Reply:
Ah got it. I should have read more closely.

Reply:
Thanks!